payu/experiment.py

```diff
@@ -1050,6 +1050,10 @@ def get_restarts_to_prune(self,
                           'model. To use integer based restart pruning, '
                           'set restart_freq to an integer value.')
                     raise
+                except FileNotFoundError as e:
+                    print(f'payu: warning: Ignoring {restart} from date-based '
+                          f'restart pruning. Error: {e}')
+                    continue
                 except Exception:
```

Comment:
Hope you don't hate me ... but it's always good practice if you're fixing a specific bug to write a test for it, confirm the test fails and then implement the fix and confirm the test passes.

Now maybe it isn't necessary in this case, but it is a good habit to get into, to confirm

1. the bug is what you think it is (by writing a test that trips it)
2. the code fix does what you think it's doing, i.e. fixing the bug

I have learnt this the hard way.

Reply:
Ok I had tested it with some test model configuration and modifying local archive files. But I just added in an small test

Reply:
Awesome. Thanks.